gitlab-ci-runner/runner/Build.cs

```diff
@@ -390,12 +383,12 @@ private string checkoutCmd()
         /// <returns>Clone CMD</returns>
         private string cloneCmd()
         {
-            String sCmd = "";
-
-            // Git Clone
-            sCmd += "git clone " + buildInfo.repo_url + " " + sProjectDir;
+            String[] sCmd = { 

```

Comment:
you should refactor this pattern into a command object.

API like:

    return
      Command.New()
        .Add("git clone " + buildInfo.repo_url + " " + sProjectDir)
        .Add("git checkout " + buildInfo.sha)
        .ToString();